libraries/AP_GPS/AP_GPS_FEMTO.cpp

```diff
@@ -0,0 +1,310 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+//  Femtomes GPS driver for ArduPilot.
+//  Code by Rui Zheng <ruizheng@femtomes.com>
+
+
+#include "AP_GPS.h"
+#include "AP_GPS_FEMTO.h"
+#include <AP_Logger/AP_Logger.h>
+
+extern const AP_HAL::HAL& hal;
+
+#define FEMTO_DEBUGGING 0
+
+#define FEMTO_MSG_ID_UAVGPS 		8001
```

Comment:
```suggestion
static const uint16_t FEMTO_MSG_ID_UAVGPS = 8001;
static const uint8_t FEMTO_MSG_ID_BESTPOS = 42;
```
In private in the class

Reply:
ok